addons/kafka/templates/cmpd-broker-27.yaml

```diff
@@ -0,0 +1,247 @@
+          name: advertised-listener
+          optional: true
+          port:
+            name: broker
+            option: Optional
+  systemAccounts:
+    - name: client
+      secretRef:
+        name: {{ include "kafka.defaultClientSystemAccountSecretName" . }}
+        namespace: {{ .Release.Namespace }}
+    - name: admin
+      secretRef:
+        name: {{ include "kafka.defaultSuperUserSystemAccountSecretName" . }}
+        namespace: {{ .Release.Namespace }}
+  ## serial is not used because rsm currently does not support kafka's role detection. The lack of role label during restart will affect the pod restart.
+  updateStrategy: BestEffortParallel
+  configs:
+    - name: broker-env
+      templateRef: {{ include "kafka2.brokerEnvTplName" . }}
+      constraintRef: {{ include "kafka2.envConstraintName" . }}
+      namespace: {{ .Release.Namespace }}
+      keys:
+        - conf
+      injectEnvTo:
+        - kafka
+      volumeName: kafka-broker-env
+    - name: kafka-configuration-tpl
+      constraintRef: {{ include "kafka2.configConstraintName" . }}
+      templateRef: {{ include "kafka2.configurationTplName" . }}
+      volumeName: kafka-config
+      namespace: {{ .Release.Namespace }}
+    - name: kafka-jmx-configuration-tpl
+      templateRef: {{ include "kafka.jmxConfigurationTplName" . }}
+      volumeName: jmx-config
+      namespace: {{ .Release.Namespace }}
+  scripts:
+    - name: kafka-scripts-tpl
+      templateRef: {{ include "kafka2.serverScriptsTplName" . }}
+      volumeName: scripts
+      namespace: {{ .Release.Namespace }}
+      defaultMode: 0755
+    - name: kafka-tools-scripts-tpl
+      templateRef: {{ include "kafka.toolsScriptsTplName" . }}
+      volumeName: tools
+      namespace: {{ .Release.Namespace }}
+      defaultMode: 0755
+  runtime:
+    securityContext:
+      fsGroup: 1001
+    containers:
+      - name: kafka
+        image: {{ .Values.images.registry | default "docker.io" }}/{{ .Values.images.kafka.repository }}:{{ default .Values.images.kafka2.tag }}
+        imagePullPolicy: {{ default "IfNotPresent" .Values.images.pullPolicy }}
+        securityContext:
+          allowPrivilegeEscalation: false
+          runAsNonRoot: true
+          runAsUser: 1001
+        command:
+{{/*          - sleep*/}}
+{{/*          - infinity*/}}
+          - /scripts/kafka-server-setup.sh
+        env:
+          - name: BITNAMI_DEBUG
+            value: {{ .Values.debugEnabled | quote }}
+          - name: MY_POD_IP
+            valueFrom:
+              fieldRef:
+                apiVersion: v1
+                fieldPath: status.podIP
+          - name: MY_POD_NAME
+            valueFrom:
+              fieldRef:
+                apiVersion: v1
+                fieldPath: metadata.name
+          - name: MY_POD_HOST_IP
+            valueFrom:
+              fieldRef:
+                apiVersion: v1
+                fieldPath: status.hostIP
+          - name: KAFKA_CFG_PROCESS_ROLES
```

Comment:
'Process role' is a concept in the kraft architecture. For 'cmpd-broker' of Kafka 2.7, this component is actually a certain 'broker', and this environment variable is not needed.

Reply:
fixed